Intermittent test failure :mail:protocols:pop3:test - certificate error when trying to connect should return CertificateError FAILED

### Checklist

- [x] I have used the search function to see if someone else has already submitted the same bug report.
- [x] I will describe the problem with as much detail as possible.

### App

Thunderbird for Android

### App version

n/a

### Where did you get the app from?

None

### Android version

n/a

### Device model

_No response_

### Steps to reproduce

See daily build:
https://github.com/thunderbird/thunderbird-android/actions/runs/19895307570/job/57024324629

### Expected behavior

This test has been failing intermittently in daily builds. It's failed 2 days in a row now.

```
> Task :mail:protocols:pop3:test FAILED

Pop3ServerSettingsValidatorTest > certificate error when trying to connect should return CertificateError FAILED
    java.lang.AssertionError: Connection open when is shouldn't be
        at com.fsck.k9.mail.store.pop3.MockPop3Server.verifyConnectionClosed(MockPop3Server.java:165)
        at com.fsck.k9.mail.store.pop3.Pop3ServerSettingsValidatorTest.certificate error when trying to connect should return CertificateError(Pop3ServerSettingsValidatorTest.kt:260)
```

### Actual behavior

n/a

### Logs

_No response_

--- a/mail/protocols/pop3/src/test/java/com/fsck/k9/mail/store/pop3/MockPop3Server.java
+++ b/mail/protocols/pop3/src/test/java/com/fsck/k9/mail/store/pop3/MockPop3Server.java
@@ -157,12 +157,13 @@ public class MockPop3Server {
         checkServerRunning();
 
         try {
-            waitForConnectionClosed.await(300L, TimeUnit.MILLISECONDS);
+            // CI can be slower; allow more time for the server thread to observe the close and signal
+            waitForConnectionClosed.await(1000L, TimeUnit.MILLISECONDS);
         } catch (InterruptedException ignored) {
         }
 
         if (!mockServerThread.isClientConnectionClosed()) {
-            throw new AssertionError("Connection open when is shouldn't be");
+            throw new AssertionError("Connection open when it shouldn't be");
         }
     }
 
@@ -283,7 +284,6 @@ public class MockPop3Server {
                     readAdditionalCommands();
                 }
 
-                waitForConnectionClosed.countDown();
             } catch (UnexpectedCommandException e) {
                 unexpectedCommandException = e;
             } catch (IOException e) {
@@ -293,11 +293,14 @@ public class MockPop3Server {
             } catch (KeyStoreException | CertificateException | UnrecoverableKeyException |
                     NoSuchAlgorithmException | KeyManagementException e) {
                 throw new RuntimeException(e);
+            } finally {
+                // Ensure we always close the socket and signal that the connection has been closed,
+                // even when we exit due to exceptions (e.g., client-initiated close).
+                IOUtils.closeQuietly(socket);
+                waitForConnectionClosed.countDown();
+
+                logger.log("Exiting");
             }
-
-            IOUtils.closeQuietly(socket);
-
-            logger.log("Exiting");
         }
 
         private void handleInteractions(Socket socket) throws IOException, KeyStoreException,

